### PR TITLE
Parse system transfer messages

### PR DESCRIPTION
**Pull request by garious**

### CODE REVIEW

**Comment by garious** [comment]
@CriesofCarrots, can you take a peek at `test_parse_system_transfer_instructions`?  Does that code seem readable enough? We're going to have to repeat it lots.

File: libsol/system_instruction.c [hunk 1]
```diff
@@ -16,26 +17,48 @@ enum SystemInstructionKind {
...
+
+// Returns 0, the fee payer, and system transfer info if provided a transfer message, otherwise non-zero.
+int parse_system_transfer_instructions(Parser* parser, MessageHeader* header, SystemTransferInfo* info) {
+    BAIL_IF(header->instructions_length != 1);
+
+    Instruction instruction;
+    BAIL_IF(parse_instruction(parser, &instruction));
+
+    Pubkey* program_id = &header->pubkeys[instruction.program_id_index];
+    Pubkey system_program_id = {0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0};
```
Comment (CriesofCarrots) [inline libsol/system_instruction.c]: Define as a constant somewhere? I assume you're going to need this in other instruction types.

File: libsol/system_instruction_test.c [hunk 2]
```diff
@@ -2,7 +2,24 @@
...
+    assert(parse_message_header(&parser, &header) == 0);
+
+    Pubkey* fee_payer_pubkey = &header.pubkeys[0];
+    SystemTransferInfo info;
+    assert(parse_system_transfer_instructions(&parser, &header, &info) == 0);
+    assert(parser.buffer_length == 0);
+    assert(info.lamports == 42);
+    assert(memcmp(fee_payer_pubkey, info.from, PUBKEY_SIZE) == 0);
+}
+
```
Comment (CriesofCarrots) [inline libsol/system_instruction_test.c]: Add a test where fee_payer and from are different?

File: libsol/system_instruction.c [hunk 3]
```diff
@@ -16,26 +17,48 @@ enum SystemInstructionKind {
...
 
     uint8_t to_index = instruction->accounts[1];
     BAIL_IF(to_index >= pubkeys_length);
-    *to = &pubkeys[to_index];
+    info->to = &pubkeys[to_index];
+
+    return 0;
+}
+
+// Returns 0, the fee payer, and system transfer info if provided a transfer message, otherwise non-zero.
```
Comment (CriesofCarrots) [inline libsol/system_instruction.c]: How does this return the fee payer? Just tacitly as header->pubkeys[0]?
Perhaps update this comment to be more clear.

Comment (garious) [reply libsol/system_instruction.c]: I failed to update the comment after changing the code. I'll fix that!

File: libsol/system_instruction.c (hunk 1 above)
Comment (garious) [reply libsol/system_instruction.c]: I'm afraid! There's a bunch of linking pitfalls in the embedded environment, so I'm starting with the thing that'll most likely work and then refactor a little later.

File: libsol/system_instruction.c (hunk 3 above)
Comment (garious) [reply libsol/system_instruction.c]: Fixed in f3c1309902022404aae86c28f56df5659f7985b8

File: libsol/system_instruction_test.c (hunk 2 above)
Comment (garious) [reply libsol/system_instruction_test.c]: Added in 
bf34d34

**Pull request merged**